apps/system/dat/dat_main.c

```diff
@@ -0,0 +1,326 @@
+
+	tc_fd = open(OS_API_TEST_DRVPATH, O_WRONLY);
+	if (tc_fd < 0) {
+		printf("FAILED to open OS API test driver %s\n", OS_API_TEST_DRVPATH);
+		return;
+	}
+
+	ret = ioctl(tc_fd, TESTIOC_GET_APP_COUNT, 0);
+	if (ret	< 0) {
+		printf("ERROR executing Application ioctl!! ERROR code = %d\n", errno);
+		return;
+	}
+
+	printf("Number of binaries : %d\n\n", ret);
+
+	/* Calculate app running time */
+	printf("Application Runtime Information\n");
+
+	clock_t ticktime;
+#ifdef CONFIG_SYSTIME_TIME64
+	uint64_t sec;
+	uint64_t start_sec;
+#else
+	uint32_t sec;
+	uint32_t start_sec;
+#endif
+	unsigned int csec;
+	unsigned int start_csec;
+
+	ticktime = clock();
+
+	/* Convert the system up time to seconds + hundredths of second */
+	sec = ticktime / CLOCKS_PER_SEC;
+	csec = (100 * (ticktime % CLOCKS_PER_SEC) + (CLOCKS_PER_SEC / 2)) / CLOCKS_PER_SEC;
+
+	/* Make sure that rounding did not force the hundredths of a second above 99 */
+	if (csec > 99) {
+		sec++;
+		csec -= 100;
+	}
+
+	/* Retrieve app1 start timings */
+
+	start_sec = ioctl(tc_fd, TESTIOC_GET_APP_START_TIME_SEC, 1);
+	if (start_sec < 0) {
+		printf("ERROR executing Application ioctl!! ERROR code = %d\n", errno);
+	}
+
+	start_csec = ioctl(tc_fd, TESTIOC_GET_APP_START_TIME_CSEC, 1);
+	if (start_csec < 0) {
+		printf("ERROR executing Application ioctl!! ERROR code = %d\n", errno);
+	}
+
+	printf("App[1] runtime :  %u.%u\n", sec - start_sec, csec - start_csec);
+
+	/* Retrieve app2 start timings */
+
+	start_sec = ioctl(tc_fd, TESTIOC_GET_APP_START_TIME_SEC, 2);
+	if (start_sec < 0) {
+		printf("ERROR executing Application ioctl!! ERROR code = %d\n", errno);
+	}
+
+	start_csec = ioctl(tc_fd, TESTIOC_GET_APP_START_TIME_CSEC, 2);
+	if (start_csec < 0) {
+		printf("ERROR executing Application ioctl!! ERROR code = %d\n", errno);
+	}
+
+	printf("App[2] runtime :  %u.%u\n", sec - start_sec, csec - start_csec);
+
+#ifdef CONFIG_SAVE_BIN_SECTION_ADDR
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	/* Display app memory address info */
+	printf("\nApplication Memory Information\n");
+	ret = ioctl(tc_fd, TESTIOC_GET_APP_MEMORY_ADDR, 0);
+	if (ret	< 0) {
+		printf("ERROR executing Application ioctl!! ERROR code = %d\n", errno);
+	}
+#endif
+#endif
+	printf("End of Application Information\n");
+	close(tc_fd);
+}
+#endif
+
+/****************************************************************************
+ * dat_main
+ ****************************************************************************/
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int dat_main(int argc, char *argv[])
+#endif
+{
+	if (argc < 2 || argc > 2) {
+		show_usage();
+	} else {
+
+		if (strncmp(argv[1], "apps", 5) == 0) {
+#ifdef CONFIG_APP_BINARY_SEPARATION
+			/* apps : show application information (number, heap, memory) */
+			dat_app_info();
+#endif
+		} else if (strncmp(argv[1], "mpu", 4) == 0) {
+#ifdef CONFIG_ARM_MPU
+			/* mpu : show number of MPU regions */
+			dat_mpu_info();
+#endif
```

Comment:
When the config is disabled, this will return nothing. No result, no error. This is not a good usage. When the config is disabled, let's remove the `mpu` argu in the USAGES and return `invalid option`.